tagstudio/src/qt/ts_qt.py

```diff
@@ -947,7 +947,7 @@ def run_macro(self, name: str, entry_id: int):
         """Runs a specific Macro on an Entry given a Macro name."""
         entry = self.lib.get_entry(entry_id)
         path = self.lib.library_dir / entry.path / entry.filename
-        source = entry.path.parts[0]
+        source = "" if entry.path == Path(".") else entry.path.parts[0]
```

Comment:
This would ensure that the source folder remains case-insensitive. Preferably, this could be done within the `get_gdl_sidecar()` method instead to sanitize any input that comes in (ex. creating `source_: str = source.lower()` in there).
```suggestion
        source = "" if entry.path == Path(".") else entry.path.parts[0].lower()
```